packages/ui-date-time-input/src/DateTimeInput/index.tsx

```diff
@@ -0,0 +1,559 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2018 - present Instructure, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+import React, { Component, SyntheticEvent, MouseEvent } from 'react'
+import { Locale, DateTime, ApplyLocaleContext } from '@instructure/ui-i18n'
+import type { Moment } from '@instructure/ui-i18n'
+import { FormFieldGroup } from '@instructure/ui-form-field'
+import type { FormMessage } from '@instructure/ui-form-field'
+
+import { DateInput } from '@instructure/ui-date-input'
+import { TimeSelect } from '@instructure/ui-time-select'
+import { Calendar } from '@instructure/ui-calendar'
+import { testable } from '@instructure/ui-testable'
+import { AccessibleContent } from '@instructure/ui-a11y-content'
+import { IconButton } from '@instructure/ui-buttons'
+import {
+  IconArrowOpenEndSolid,
+  IconArrowOpenStartSolid
+} from '@instructure/ui-icons'
+import { debounce } from '@instructure/debounce'
+import type { DateTimeInputProps, DateTimeInputState } from './props'
+import { propTypes, allowedProps } from './props'
+
+/**
+---
+category: components
+---
+@tsProps
+**/
+@testable()
+class DateTimeInput extends Component<DateTimeInputProps, DateTimeInputState> {
```

Comment:
Nah, I meant a migration guide here in V8, that explains why this component was "revived" and how it can be migrated from the old version. I'm sure the Canvas team would be super glad to have a guide for it.

Reply:
I've added some docs about this new component, but did not wrote a migration guide because likely we will do the migration and it only needs to be done for Canvas